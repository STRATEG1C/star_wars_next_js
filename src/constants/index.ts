export const API_URL = 'http://swapi.dev';

export const API_PEOPLE = `${API_URL}/api/people/`;

export const API_RANDOM_IMAGE = 'https://picsum.photos';

export const API_RANDOM_PEOPLE = 'https://randomuser.me/api';

export const DEFAULT_NUMBER_OF_POSTS = 6;
