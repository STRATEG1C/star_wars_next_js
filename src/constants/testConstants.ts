import { IPost } from '../components/PostsList/types';

export const testPost: IPost = {
    thumbnail: '',
    author: {
        avatar: 'string',
        name: 'Dart Veider',
        specie: 'Human',
        home: 'Star'
    }
};

export const testPostList: IPost[] = [
    {
        thumbnail: '',
        author: {
            avatar: 'string',
            name: 'Dart Veider',
            specie: 'Human',
            home: 'Star'
        }
    },
    {
        thumbnail: '',
        author: {
            avatar: 'string',
            name: 'Obivan',
            specie: 'Human',
            home: 'Star'
        }
    }
];
