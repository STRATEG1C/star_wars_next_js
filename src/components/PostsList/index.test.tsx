import React from 'react';
import { render } from 'enzyme';

import { testPostList } from '../../constants/testConstants';

import PostList from './index';

describe('Should render PostList', () => {
    let component;
    const onRemovePost = jest.fn();

    beforeEach(() => {
        component = render(
            <PostList
                posts={testPostList}
                onRemovePost={onRemovePost}
            />
        );
    })

    it('Should render posts', () => {
        expect(component.find('.post').length).toBe(testPostList.length);
        expect(component).toMatchSnapshot();
    });
})
