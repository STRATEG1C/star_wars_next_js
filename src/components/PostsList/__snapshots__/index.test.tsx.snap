// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render PostList Should render posts 1`] = `
<div
  class="post-list flex flex-wrap"
>
  <article
    class="post my-1 px-1 w-full md:w-1/2 lg:my-4 lg:px-4 lg:w-1/3"
  >
    <div
      class="w-full flex justify-between"
    >
      <div
        class="flex"
      >
        <div
          class="rounded-full h-10 w-10 bg-gray-500 flex items-center justify-between overflow-hidden"
        >
          <img
            alt="profilepic"
            src="string"
          />
        </div>
        <div>
          <p
            class="post__author-name ml-2 font-bold text-sm"
          >
            Dart Veider
          </p>
          <p
            class="post__author-info ml-2 font-light text-sm"
          >
            Human, Star
          </p>
        </div>
      </div>
      <button>
        Remove
      </button>
    </div>
    <img
      class="w-full bg-cover mt-6 rounded"
      src="https://picsum.photos/600/400"
    />
  </article>
  <article
    class="post my-1 px-1 w-full md:w-1/2 lg:my-4 lg:px-4 lg:w-1/3"
  >
    <div
      class="w-full flex justify-between"
    >
      <div
        class="flex"
      >
        <div
          class="rounded-full h-10 w-10 bg-gray-500 flex items-center justify-between overflow-hidden"
        >
          <img
            alt="profilepic"
            src="string"
          />
        </div>
        <div>
          <p
            class="post__author-name ml-2 font-bold text-sm"
          >
            Obivan
          </p>
          <p
            class="post__author-info ml-2 font-light text-sm"
          >
            Human, Star
          </p>
        </div>
      </div>
      <button>
        Remove
      </button>
    </div>
    <img
      class="w-full bg-cover mt-6 rounded"
      src="https://picsum.photos/600/400"
    />
  </article>
</div>
`;
